apps/frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,24 @@
+import { type FC, type ReactNode } from "react";
+import { Navigate } from "react-router-dom";
+
+import { AppRoute } from "~/libs/enums/app-route.enum.js";
+import { DataStatus } from "~/libs/enums/data-status.enum.js";
+import { useAppSelector } from "~/libs/hooks/hooks.js";
+
+type ProtectedRouteProperties = {
+	children: ReactNode;
+};
+
+const ProtectedRoute: FC<ProtectedRouteProperties> = ({
+	children,
+}: ProtectedRouteProperties): JSX.Element => {
+	const authStatus = useAppSelector((state) => state.auth.dataStatus);
+
+	return authStatus === DataStatus.FULFILLED ? (
```

Comment:
Let's check it by user presence, not status. Because there could be other conditions when the status is fulfilled. We will store it in `auth.authenticatedUser` in store

Reply:
Yes, you can, use SignUpResponseDto as a type